From e2e62655bf65c3d90cd5e71741c064fb023c4f4f Mon Sep 17 00:00:00 2001
From: Ahmad Abbas <ahmada@marvell.com>
Date: Wed, 6 May 2015 12:59:41 +0300
Subject: [PATCH 0117/2241] fdt: spi: enc28j60 added support for the enc28j60
 in the fdt

	added support for the enc28j60 driver to work using fdt.
	the new device works on spi 0, chip select 1 with spi mode 0

Signed-off-by: Ahmad Abbas <ahmada@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/19028
Tested-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Signed-off-by: Neta Zur <neta@marvell.com>

Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>

Conflicts:
	drivers/net/ethernet/microchip/enc28j60.c

Change-Id: Ib04e98674f70509047d9e265f1257a887f32bbba
Reviewed-on: http://vgitil04.il.marvell.com:8080/27569
Reviewed-by: Lior Amsalem <alior@marvell.com>
Tested-by: Lior Amsalem <alior@marvell.com>
---
 arch/arm/boot/dts/armada-385-db-ap.dts    | 5 +++++
 drivers/net/ethernet/microchip/enc28j60.c | 6 ++++++
 2 files changed, 11 insertions(+)

diff --git a/arch/arm/boot/dts/armada-385-db-ap.dts b/arch/arm/boot/dts/armada-385-db-ap.dts
index acd5b15..4282e06 100644
--- a/arch/arm/boot/dts/armada-385-db-ap.dts
+++ b/arch/arm/boot/dts/armada-385-db-ap.dts
@@ -76,6 +76,11 @@
 					reg = <0>; /* Chip select 0 */
 					spi-max-frequency = <54000000>;
 				};
+				ethernet: enc28j60@0 {
+					compatible = "microchip,enc28j60";
+					spi-max-frequency = <20000000>;
+					reg = <1>;
+				};
 			};
 
 			i2c0: i2c@11000 {
diff --git a/drivers/net/ethernet/microchip/enc28j60.c b/drivers/net/ethernet/microchip/enc28j60.c
index 86ea17e..3407545 100644
--- a/drivers/net/ethernet/microchip/enc28j60.c
+++ b/drivers/net/ethernet/microchip/enc28j60.c
@@ -1630,9 +1630,15 @@ static int enc28j60_remove(struct spi_device *spi)
 	return 0;
 }
 
+static struct of_device_id enc28j60_dt_ids[] = {
+	{ .compatible = "microchip,enc28j60", },
+	{}
+};
+
 static struct spi_driver enc28j60_driver = {
 	.driver = {
 		   .name = DRV_NAME,
+		   .of_match_table = enc28j60_dt_ids,
 	 },
 	.probe = enc28j60_probe,
 	.remove = enc28j60_remove,
-- 
2.7.4

